Sources/PubNub/Helpers/Crypto/CryptorModule.swift

```diff
@@ -0,0 +1,320 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Represents the result of stream encryption
+public struct EncryptedStreamResult {
+  /// Encoded stream you can read from
+  public let stream: InputStream
+  /// Content length of encoded stream
+  public let contentLength: Int
+  
+  public init(stream: InputStream, contentLength: Int) {
+    self.stream = stream
+    self.contentLength = contentLength
+  }
+}
+
+/// Object capable of encryption/decryption.
+///
+/// - Important: Replaces obsolete [Crypto](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/Crypto.swift#L32)
+public struct CryptorModule {
+  private let defaultCryptor: Cryptor
+  private let cryptors: [Cryptor]
+  private let legacyCryptorId: CryptorId = []
+  private let defaultStringEncoding: String.Encoding
+  
+  typealias Base64EncodedString = String
+  
+  /// Initializes `CryptorModule` with custom ``Cryptor`` objects capable of encryption and decryption
+  ///
+  /// Use this constructor if you would like to provide **custom** objects for decryption and encryption and don't want to use PubNub's built-in `Cryptors`.
+  /// Otherwise, refer to convenience static factory methods such as ``aesCbcCryptoModule(with:withRandomIV:)``
+  /// and ``legacyCryptoModule(with:withRandomIV:)`` that returns `CryptorModule` configured for you.
+  ///
+  /// - Parameters:
+  ///   - default: Primary ``Cryptor`` instance used for encryption and decryption
+  ///   - cryptors: An optional list of ``Cryptor`` instances which older messages/files were encoded
+  ///   - encoding: Default String encoding used when publishing new messages
+  public init(default cryptor: Cryptor, cryptors: [Cryptor] = [], encoding: String.Encoding = .utf8) {
+    self.defaultCryptor = cryptor
+    self.cryptors = cryptors
+    self.defaultStringEncoding = encoding
+  }
+  
+  /// Encrypts the given `Data` object
+  /// 
+  /// - Parameters:
+  ///   - data: Data to encrypt
+  /// - Returns: A success, storing encrypted `Data` if operation succeeds. Otherwise, a failure storing `PubNubError` is returned
+  public func encrypt(data: Data) -> Result<Data, PubNubError> {
+    guard !data.isEmpty else {
+      return .failure(PubNubError(
+        .encryptionError,
+        additional: ["Cannot encrypt empty Data"])
+      )
+    }
+    return defaultCryptor.encrypt(data: data).map {
+      if defaultCryptor.id == LegacyCryptor.ID {
+        return $0.data
+      } else {
+        return CryptorHeader.v1(
+          cryptorId: defaultCryptor.id,
+          data: $0.metadata
+        ).asData() + $0.data
```

Comment:
There are some notes further in PR about `CryptorHeader` that it should represent only information about cryptor, but actual data retrieved and composed here:
```suggestion
        return CryptorHeader.v1(
          cryptorId: defaultCryptor.id,
          dataSize: $0.metadata.count
        ).asData() + $0.metadata + $0.data
```